tests/unit/test-util/test.artifacts.js

```diff
@@ -55,4 +56,38 @@ describe('prepareArtifactsDir', () => {
     }
   ));
 
+  it('throws error when lacking writing permissions', () => withTempDir(
+    (tmpDir) => {
+      const tmpPath = path.join(tmpDir.path(), 'build');
+      return fs.mkdir(tmpPath, '0622').then(() => {
+        const artifactsDir = path.join(tmpPath, 'artifacts');
+        return prepareArtifactsDir(artifactsDir)
+          .then(makeSureItFails())
+          .catch(onlyInstancesOf(UsageError, (error) => {
+            assert.match(error.message, /lack permissions/);
+          }));
+      });
+    }
+  ));
+
+  it('throws error when creating writing permissions', () => withTempDir(
+    (tmpDir) => {
+      class ErrorWithCode extends Error {
+        code: string;
+        constructor() {
+          super('some error');
+          this.code = 'EACCES';
+        }
+      }
+
+      const tmpPath = path.join(tmpDir.path(), 'build');
+      fs.mkdir = sinon.spy(() => Promise.reject(new ErrorWithCode()));
```

Comment:
This is risky, if we change this module function and the spy is actually accessed by other modules, we are going to potentially affect other tests, and it is going to potentially generate errors that can be very hard to be debugged (because a fake installed by a test can generate a failure in another test executed after it).

If we want to do this kind of tests (and it is actually a good idea, even better than creating the actually files and dirs on the filesystem), we should use the pattern that we already use for the other similar scenarios:

- the tested component (e.g. the `prepareArtifactsDir` is this case) will take an additional parameter which contains the dependencies (the `fs` module in this case), with the related properties defaulted to the default implementation (the `fs` property imported from the `"mz"` npm module in this case)
- in the real code, the additional parameter is not passed (and the defaults are used)
- in the test code, we pass the additional parameter to inject the fake implementation that behaves as needed by the test case

(e.g. basically the same pattern that you are using for #674)